src/requests/consumption.js

```diff
@@ -0,0 +1,744 @@
+	}
+}
+
+const duplicateResources = async (resourceDetails) => {
+	let projectTemplateIds = []
+	let solutionTemplateIds = []
+	resourceDetails.forEach((resource) => {
+		if (resource.type == common.PROJECT) projectTemplateIds.push(ObjectId(resource._id))
+		else solutionTemplateIds.push(ObjectId(resource._id))
+	})
+
+	// handling only project creation now. Make changes here for observation , survey etc...
+	if (projectTemplateIds.length > 0) {
+		const projectsCollection = mongoDb.collection(COLLECTIONS.PROJECT_TEMPLATES)
+		const projectTemplates = await projectsCollection
+			.find({
+				_id: {
+					$in: projectTemplateIds,
+				},
+			})
+			.toArray()
+
+		//templateProjectsTaskMap = {
+		// 	projectExternalId : [ list of last ids]
+		// }
+		let templateProjectsTaskMap = {}
+		let templateProjects = []
+		let templateTaskIds = []
+		let duplicateTasks = []
+
+		//taskMap = {
+		// 	projectTaskId : duplicateProjectTaskId
+		// }
+		let taskMap = {}
+
+		if (projectTemplates) {
+			projectTemplates.forEach((project) => {
+				project.externalId = project.externalId + common.SUFFIX_CHILD
+				delete project._id
+				project.updatedAt = new Date()
+				project.createdAt = new Date()
+				project.isReusable = false
+				templateProjectsTaskMap[project.externalId] = project.tasks
+				templateProjects.push(project)
+			})
+
+			Object.keys(templateProjectsTaskMap).forEach(async (projectExtId) => {
+				templateTaskIds = [...templateTaskIds, ...templateProjectsTaskMap[projectExtId]]
+			})
+			templateTaskIds = [...new Set(templateTaskIds)]
+			const projectsTaskCollection = mongoDb.collection(COLLECTIONS.TASKS)
+			const projectsTasksDetails = await projectsTaskCollection
+				.find({
+					_id: {
+						$in: templateTaskIds,
+					},
+				})
+				.toArray()
+
+			projectsTasksDetails.forEach((projectTask) => {
+				projectTask.externalId = utils.generateUniqueId()
+				taskMap[projectTask._id] = projectTask.externalId
+				projectTask.updatedAt = new Date()
+				projectTask.createdAt = new Date()
+				delete projectTask._id
+				duplicateTasks.push(projectTask)
+			})
+
+			await projectsTaskCollection.insertMany(duplicateTasks)
+
+			const projectsTasksDetailsAfterInsert = await projectsTaskCollection
+				.find({
+					externalId: {
+						$in: duplicateTasks.map((tasks) => tasks.externalId),
+					},
+				})
+				.toArray()
+
+			taskMap = _.mapValues(taskMap, (externalId) => {
+				// Find the corresponding object from projectsTasksDetailsAfterInsert
+				const task = _.find(projectsTasksDetailsAfterInsert, { externalId: externalId })
+
+				// If found, replace externalId with _id; otherwise, keep the externalId
+				return task ? ObjectId(task._id) : externalId
+			})
+
+			templateProjects.forEach((project) => {
+				let projectTasks = []
+				project.tasks.forEach((task) => {
+					projectTasks.push(taskMap[task])
+				})
+				project.tasks = projectTasks
+			})
+
+			await projectsCollection.insertMany(templateProjects)
```

Comment:
@adithyadinesh0412  why do we need  insert many here ? also in the second time push, can we update the duplicate template instead of creating new